proxy/http/HttpSM.cc

```diff
@@ -782,9 +782,11 @@ HttpSM::state_read_client_request_header(int event, void *data)
         ua_entry->write_buffer    = new_MIOBuffer(alloc_index);
         IOBufferReader *buf_start = ua_entry->write_buffer->alloc_reader();
 
+        t_state.is_100_continue_request = true;
+
         DebugSM("http_seq", "send 100 Continue response to client");
-        int64_t nbytes = ua_entry->write_buffer->write(str_100_continue_response, len_100_continue_response);
-        ua_session->do_io_write(netvc, nbytes, buf_start);
+        int64_t nbytes      = ua_entry->write_buffer->write(str_100_continue_response, len_100_continue_response);
+        ua_entry->write_vio = ua_session->do_io_write(this, nbytes, buf_start);
```

Comment:
Save the vio returned from do_io_write to ua_entry->write_vio. 
It makes HttpSM::main_handler calls ua_entry->vc_handler that is currently point to HttpSM::state_watch_for_client_abort.